src/Toolkit/src/Command/BuildRegistryCommand.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Toolkit\Command;
+
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Finder\Finder;
+use Symfony\UX\Toolkit\Compiler\RegistryCompiler;
+use Symfony\UX\Toolkit\Registry\Registry;
+use Symfony\UX\Toolkit\Registry\RegistryItemFactory;
+
+/**
+ * @author Jean-François Lépine
+ * @author Hugo Alliaume <hugo@alliau.me>
+ *
+ * @internal
+ */
+#[AsCommand(
+    name: 'ux:toolkit:build-registry',
+    description: 'This command allows to distribute your components, and build the registry.'
+)]
+class BuildRegistryCommand extends Command
+{
+    public function __construct(
+        private readonly RegistryCompiler $compiler,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addOption(
+                'template-directory',
+                't',
+                InputOption::VALUE_REQUIRED,
+                'The directory where the templates are stored.',
+                'templates'
+            )
+            ->addOption(
+                'destination',
+                'r',
+                InputOption::VALUE_REQUIRED,
+                'The directory where the registry will be stored.',
+                'registry'
+            )
+            ->addArgument('name', InputArgument::OPTIONAL, 'The name of the component.', '')
+            ->addArgument('homepage', InputArgument::OPTIONAL, 'The homepage of the component.', 'https://...')
```

Comment:
Those are not the name or homepage of the component, but of the registry.

Anyway, maybe we can have a configuration file `templates/<theme>/manifest.json` that will already contains those data?